tests/test_parser.py

```diff
@@ -229,6 +229,144 @@ def g():
         )
 
 
+def test_self_recursion_parser_opt():
+    def f():
```

Comment:
For coverage mark this function with `# pragma: no cover` (and all the other tests functions)